From ce05db675a5aac8390148df67a00266037e5076a Mon Sep 17 00:00:00 2001
From: He Junyan <junyan.he@intel.com>
Date: Mon, 1 May 2023 23:34:58 +0800
Subject: [PATCH 11/14] va: Set the AV1 reference list number correctly for iHD
 driver

The iHD driver messes it up and we should delete this hack when bug
https://github.com/intel/media-driver/issues/1620
is fixed.
---
 subprojects/gst-plugins-bad/sys/va/gstvaencoder.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/subprojects/gst-plugins-bad/sys/va/gstvaencoder.c b/subprojects/gst-plugins-bad/sys/va/gstvaencoder.c
index d1e8ed1d0e..b524b00cf2 100644
--- a/subprojects/gst-plugins-bad/sys/va/gstvaencoder.c
+++ b/subprojects/gst-plugins-bad/sys/va/gstvaencoder.c
@@ -672,6 +672,15 @@ gst_va_encoder_get_max_num_reference (GstVaEncoder * self,
   if (list1)
     *list1 = (attrib.value >> 16) & 0xffff;
 
+  if ((gst_va_display_get_implementation (self->display) ==
+          GST_VA_IMPLEMENTATION_INTEL_IHD) &&
+      (profile == VAProfileAV1Profile0 || profile == VAProfileAV1Profile1)) {
+    if (list0)
+      *list0 = (attrib.value >> 8) & 0xff;
+    if (list1)
+      *list1 = (attrib.value >> 16) & 0xff;
+  }
+
   return TRUE;
 }
 
-- 
2.25.1

